src/anemoi/training/losses/mse.py

```diff
@@ -89,3 +89,109 @@ def forward(
         # keep last dimension (variables) when summing weights
         out /= self.sum_function(self.weights[..., None].expand_as(out), axis=(0, 1, 2))
         return self.sum_function(out, axis=(0, 1, 2))
+
+
+class WeightedMSELossStretchedGrid(nn.Module):
+    """Latitude-weighted MSE loss, calculated only within or outside the limited area.
+
+    Further, the loss can be computed for the specified region (default),
+    or as the contribution to the overall loss.
+    """
+
+    def __init__(
+        self,
+        node_weights: torch.Tensor,
+        mask: torch.Tensor,
+        inside_lam: bool = True,
+        wmse_contribution: bool = False,
+        data_variances: torch.Tensor | None = None,
```

Comment:
This has been removed in favour of the `scalars`. These provide a way to use arbitary scalars on the loss calculation.
I think it may be possible to abstract `mask` into these scalars from the forecaster as well